### PR TITLE
[CRE-880, CRE-939, CRE-944] Add regression suite and new system tests, add run-e2e-regression label

### PR DESCRIPTION
**Pull request by chudilka1**
This PR represents three separate updates for the CRE system tests.
[CRE-944](https://smartcontract-it.atlassian.net/browse/CRE-944): Create label and run the CRE regression suite in CI on demand:
- Updated CI workflows to execute tests in system-tests/regression package when label `run-e2e-regression` is added to PR

[CRE-880](https://smartcontract-it.atlassian.net/browse/CRE-880): Arrange tests by moving `smoke` and `regression` suites to different packages
[CRE-939](https://smartcontract-it.atlassian.net/browse/CRE-939): Add `evm.CallContract` negative tests for invalid contract address

[CRE-944]: https://smartcontract-it.atlassian.net/browse/CRE-944?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CRE-880]: https://smartcontract-it.atlassian.net/browse/CRE-880?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CRE-939]: https://smartcontract-it.atlassian.net/browse/CRE-939?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/cre-system-tests.yaml [hunk 1]
```diff
@@ -203,11 +227,29 @@ jobs:
...
+          TEST_NAME: ${{ matrix.tests.test_name }}
+          # override chip ingress image to use remote image. We have added this env var here, instead of the "Start local CRE" step, because
+          # Beholder stack will be started only for the Beholder test
+          CHIP_INGRESS_IMAGE: ${{ secrets.AWS_ACCOUNT_ID_PROD }}.dkr.ecr.${{ secrets.QA_AWS_REGION }}.amazonaws.com/atlas-chip-ingress:bbac3c825b061546980fa9d7dc0f3e8c34347bcf
+          BILLING_PLATFORM_SERVICE_IMAGE: ${{ secrets.AWS_ACCOUNT_ID_PROD }}.dkr.ecr.${{ secrets.QA_AWS_REGION }}.amazonaws.com/billing-platform-service:v1.36.2-amd64
+          TEST_TIMEOUT: 200m
+        run: |
+          # Sleep to allow testcontainers to stop
+          # Before running it with flakeguard add this bit back: { go test... -json; exit_code=$?; ../../tools/ci/wait-for-containers-to-stop.sh 30; exit $exit_code; }
+          go test github.com/smartcontractkit/chainlink/system-tests/tests/regression/cre -v -run "^(${TEST_NAME})$" -timeout ${TEST_TIMEOUT} -count=1 -test.parallel=1 -json 2>&1 | tee /tmp/gotest.log | gotestloghelper -ci -singlepackage -hidepassingtests=false
```
Comment (jmank88) [inline .github/workflows/cre-system-tests.yaml]: I think we have deprecated gotestloghelper. You can steam the logs w/o json instead:
```suggestion
          go test github.com/smartcontractkit/chainlink/system-tests/tests/regression/cre -v -run "^(${TEST_NAME})$" -timeout ${TEST_TIMEOUT} -count=1 -test.parallel=1 2>&1 | tee /tmp/gotest.log
```

**Review by mchain0 — DISMISSED** [review]

File: .github/workflows/cre-system-tests.yaml [hunk 2]
```diff
@@ -204,13 +222,31 @@ jobs:
...
+          TEST_NAME: ${{ matrix.tests.test_name }}
+          # override chip ingress image to use remote image. We have added this env var here, instead of the "Start local CRE" step, because
+          # Beholder stack will be started only for the Beholder test
+          CHIP_INGRESS_IMAGE: ${{ secrets.AWS_ACCOUNT_ID_PROD }}.dkr.ecr.${{ secrets.QA_AWS_REGION }}.amazonaws.com/atlas-chip-ingress:bbac3c825b061546980fa9d7dc0f3e8c34347bcf
+          BILLING_PLATFORM_SERVICE_IMAGE: ${{ secrets.AWS_ACCOUNT_ID_PROD }}.dkr.ecr.${{ secrets.QA_AWS_REGION }}.amazonaws.com/billing-platform-service:v1.36.2-amd64
+          TEST_TIMEOUT: 200m
+        run: |
+          # Sleep to allow testcontainers to stop
+          # Before running it with flakeguard add this bit back: { go test... -json; exit_code=$?; ../../tools/ci/wait-for-containers-to-stop.sh 30; exit $exit_code; }
+          go test github.com/smartcontractkit/chainlink/system-tests/tests/regression/cre -v -run "^(${TEST_NAME})$" -timeout ${TEST_TIMEOUT} -count=1 -test.parallel=1 2>&1 | tee /tmp/gotest.log
```
Comment (jmank88) [inline .github/workflows/cre-system-tests.yaml:235]: Pre-existing, but just noticed: Given that the [docs](https://pkg.go.dev/cmd/go/internal/test) for `go test -list` say:
```
-list regexp
	    List tests, benchmarks, fuzz tests, or examples matching the regular
	    expression. No tests, benchmarks, fuzz tests, or examples will be run.
	    **This will only list top-level tests. No subtest or subbenchmarks will be
	    shown.**
```
Should we drop the trailing `$` from our regexes? Otherwise sub-tests are excluded.
```suggestion
          go test github.com/smartcontractkit/chainlink/system-tests/tests/regression/cre -v -run "^(${TEST_NAME})" -timeout ${TEST_TIMEOUT} -count=1 -test.parallel=1 2>&1 | tee /tmp/gotest.log
```

Comment (chudilka1) [reply .github/workflows/cre-system-tests.yaml:235]: Thanks, @jmank88, updated.

**Review by mchain0 — DISMISSED** [review]

**Review by jmank88 — DISMISSED** [review]

**Review by ilija42 — DISMISSED** [review]

**Review by george-dorin — DISMISSED** [review]

File: .github/workflows/cre-system-tests.yaml (hunk 2 above)
Comment (chudilka1) [reply .github/workflows/cre-system-tests.yaml:235]: Retracting my previous statement, the way the test matrix is defined requires ^$ to wrap the test name; otherwise, some tests will be skipped.

Comment (jmank88) [reply .github/workflows/cre-system-tests.yaml:235]: I think we must be miscommunicating. Including `$` will skip sub-tests. What effect do you see when it is excluded? You should only see more test names match, never fewer.

File: system-tests/tests/regression/cre/README.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Regression Tests
+
+These tests run on schedule or demand (label), they are time-consuming and not recommended to be triggerred on PR.
```
Comment (krehermann) [inline system-tests/tests/regression/cre/README.md]: add the value of the label here so readers can use it.

does this pr actually implement them on a schedule (say, nightly)? if so, how are failures bubbled up to engineers?

File: system-tests/tests/regression/cre/v2_cron_beholder_regression_test.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+	"github.com/smartcontractkit/chainlink-testing-framework/framework"
+
+	crontypes "github.com/smartcontractkit/chainlink/core/scripts/cre/environment/examples/workflows/v2/cron/types"
+
+	t_helpers "github.com/smartcontractkit/chainlink/system-tests/tests/test-helpers"
+	ttypes "github.com/smartcontractkit/chainlink/system-tests/tests/test-helpers/configuration"
+)
+
+// regression
```
Comment (krehermann) [inline system-tests/tests/regression/cre/v2_cron_beholder_regression_test.go:17]: what reasoning did you use to make these regression tests vs smoke tests? is there specific guidance that is documented? 

my concern is the any engineer who onboards can self-serve with docs and examples rather than long debates on slack

File: system-tests/tests/regression/cre/README.md (hunk 3 above)
Comment (chudilka1) [reply system-tests/tests/regression/cre/README.md]: Hey, Kieth, thanks, I've updated README.

We discussed with @mchain0 and @george-dorin that it makes more sense to run all tests by default and skip the `regression` on a demand/label. It should be implemented in (CRE-963)[https://smartcontract-it.atlassian.net/browse/CRE-963]

File: system-tests/tests/regression/cre/v2_cron_beholder_regression_test.go (hunk 4 above)
Comment (chudilka1) [reply system-tests/tests/regression/cre/v2_cron_beholder_regression_test.go:17]: I've added a brief description to the README on what constitutes a regression test (basically, everything that is not a `happy path` test).

File: .github/workflows/cre-system-tests.yaml (hunk 2 above)
Comment (chudilka1) [reply .github/workflows/cre-system-tests.yaml:235]: We use `t.Run` for nested tests, which are never excluded, so all the tests are eventually correctly parsed and executed compared to `$`

File: .github/workflows/cre-system-tests.yaml [hunk 5]
```diff
@@ -85,8 +97,20 @@ jobs:
...
-
-          test_names=$(go test github.com/smartcontractkit/chainlink/system-tests/tests/smoke/cre -list . | grep -v 'ok' | grep -v '^$')
+          
+          # If regression tests flag is true, include regression tests in the matrix
+          # Otherwise, only include smoke tests
+          if [[ "$WITH_REGRESSION" == 'true' ]]; then
+            echo "Including regression tests in the matrix"
+            test_names=$(go test \
+            github.com/smartcontractkit/chainlink/system-tests/tests/smoke/cre \
+            github.com/smartcontractkit/chainlink/system-tests/tests/regression/cre \
```
Comment (Tofel) [inline .github/workflows/cre-system-tests.yaml:107]: do we expect to have more suites of this kind in the future? if so, then could make more sense to pass an array of suites/folders that should be included instead of a boolean.

File: .github/workflows/cre-system-tests.yaml [hunk 6]
```diff
@@ -166,12 +190,6 @@ jobs:
         env:
           AWS_REGION: ${{ secrets.QA_AWS_REGION }}
 
-      # gotestloghelper gives us pretty test output
-      - name: Set Up gotestloghelper
-        shell: bash
-        run: go install
-          github.com/smartcontractkit/chainlink-testing-framework/tools/gotestloghelper@v1.1.1
```
Comment (Tofel) [inline .github/workflows/cre-system-tests.yaml:173]: curious, why?

File: .github/workflows/cre-system-tests.yaml [hunk 7]
```diff
@@ -204,13 +222,31 @@ jobs:
...
+      - name: Run CRE Regression system tests
+        if: ${{ inputs.with_regression }}
+        shell: bash
+        working-directory: system-tests/tests
+        env:
+          TEST_NAME: ${{ matrix.tests.test_name }}
+          # override chip ingress image to use remote image. We have added this env var here, instead of the "Start local CRE" step, because
+          # Beholder stack will be started only for the Beholder test
+          CHIP_INGRESS_IMAGE: ${{ secrets.AWS_ACCOUNT_ID_PROD }}.dkr.ecr.${{ secrets.QA_AWS_REGION }}.amazonaws.com/atlas-chip-ingress:bbac3c825b061546980fa9d7dc0f3e8c34347bcf
+          BILLING_PLATFORM_SERVICE_IMAGE: ${{ secrets.AWS_ACCOUNT_ID_PROD }}.dkr.ecr.${{ secrets.QA_AWS_REGION }}.amazonaws.com/billing-platform-service:v1.36.2-amd64
```
Comment (Tofel) [inline .github/workflows/cre-system-tests.yaml]: could we set these two env vars on the top level, so that we don't need to define them for this and smoke tests step? Or you think we might want to use different images?

File: .github/workflows/cre-system-tests.yaml [hunk 8]
```diff
@@ -204,13 +222,31 @@ jobs:
...
+          TEST_TIMEOUT: 200m
+        run: |
+          # Sleep to allow testcontainers to stop
+          # Before running it with flakeguard add this bit back: { go test... -json; exit_code=$?; ../../tools/ci/wait-for-containers-to-stop.sh 30; exit $exit_code; }
```
Comment (Tofel) [inline .github/workflows/cre-system-tests.yaml]: could you remove this comment? we don't use flakeguard anymore (and pls remove it from the smoke test step too 🙏 )

File: .github/workflows/cre-system-tests.yaml [hunk 9]
```diff
@@ -204,13 +222,31 @@ jobs:
...
+          go test github.com/smartcontractkit/chainlink/system-tests/tests/regression/cre -v -run "^(${TEST_NAME})$" -timeout ${TEST_TIMEOUT} -count=1 -test.parallel=1 2>&1 | tee /tmp/gotest.log
+
 
       - name: Show Docker containers status
         if: failure()
         shell: bash
         run: docker ps -a
-      - name: Save Docker logs
+
+      - name: Display Smoke tests Docker logs
```
Comment (Tofel) [inline .github/workflows/cre-system-tests.yaml]: this and next name is incorrect. It doesn't display the logs, it saves them in the file called `"logs/${c}.log"`.

we should never display docker logs in GH. There are TONS of them.

File: system-tests/tests/regression/cre/cre_regression_suite_test.go [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+
+// REGRESSION TESTS target edge cases, negative conditions, etc., all happy path and sanity checks should go to a `smoke` package.
+
+/*
+To execute tests locally start the local CRE first:
+Inside `core/scripts/cre/environment` directory
+ 1. Ensure the necessary capabilities (i.e. readcontract, http-trigger, http-action) are listed in the environment configuration
+ 2. Identify the appropriate topology that you want to test
+ 3. Stop and clear any existing environment: `go run . env stop -a`
+ 4. Run: `go run . env start -t <topology> && ./bin/ctf obs up` to start env + observability
```
Comment (Tofel) [inline system-tests/tests/regression/cre/cre_regression_suite_test.go]: actually this shouldn't be used anymore: `go run . env start -t <topology>`. We should rather use `CTF_CONFIGS=<path-to-your-topology-config` go run . env start`.

File: system-tests/tests/regression/cre/README.md [hunk 11]
```diff
@@ -0,0 +1,9 @@
+# Regression Tests
+
+Use `run-e2e-regression` label to run regression tests on demand demand when.
```
Comment (Tofel) [inline system-tests/tests/regression/cre/README.md]: `tests on demand demand when.` 2x `demand` and second part of the sentence missing?

File: .github/workflows/cre-system-tests.yaml (hunk 5 above)
Comment (chudilka1) [reply .github/workflows/cre-system-tests.yaml:107]: I do not envision it, basically we may consider a regression everything what is not `smoke`.

**Review by mchain0 — DISMISSED** [review]

File: system-tests/tests/go.mod [hunk 12]
```diff
@@ -48,12 +50,12 @@ require (
 	github.com/smartcontractkit/chainlink-testing-framework/lib v1.54.5
 	github.com/smartcontractkit/chainlink-testing-framework/seth v1.51.2
 	github.com/smartcontractkit/chainlink-testing-framework/wasp v1.51.1
-	github.com/smartcontractkit/chainlink/core/scripts/cre/environment/examples/workflows/v1/proof-of-reserve/cron-based v0.0.0-20250826151008-ae5ec0ee6f2c
-	github.com/smartcontractkit/chainlink/core/scripts/cre/environment/examples/workflows/v2/cron v0.0.0-20250916160605-4f051f3d3f51
+	github.com/smartcontractkit/chainlink/core/scripts/cre/environment/examples/workflows/v1/proof-of-reserve/cron-based v0.0.0-00010101000000-000000000000
+	github.com/smartcontractkit/chainlink/core/scripts/cre/environment/examples/workflows/v2/cron v0.0.0-00010101000000-000000000000
```
Comment (Tofel) [inline system-tests/tests/go.mod]: these pseudo-versions are wrong and might break in other repositories that import `chainlink`. Pls do the following:
- comment out replaces
- change these psuedo-versions to the latest commit SHA from this branch
- run `go mod tidy`
- restore replaces

File: system-tests/tests/go.mod [hunk 13]
```diff
@@ -48,12 +50,12 @@ require (
...
 	github.com/smartcontractkit/chainlink/deployment v0.0.0-20250906212935-a1521d24f47f
 	github.com/smartcontractkit/chainlink/system-tests/lib v0.0.0-20250826151008-ae5ec0ee6f2c
+	github.com/smartcontractkit/chainlink/system-tests/tests/regression/cre/evm/evmread-negative v0.0.0-00010101000000-000000000000
```
Comment (Tofel) [inline system-tests/tests/go.mod]: same

File: .github/workflows/cre-system-tests.yaml (hunk 6 above)
Comment (chudilka1) [reply .github/workflows/cre-system-tests.yaml:173]: According to @jmank88 , we have deprecated it, nothing has changed after removal besides the way tests are displayed.

File: .github/workflows/cre-system-tests.yaml (hunk 8 above)
Comment (chudilka1) [reply .github/workflows/cre-system-tests.yaml]: Done

**Review by george-dorin — DISMISSED** [review]

File: .github/workflows/cre-system-tests.yaml (hunk 9 above)
Comment (chudilka1) [reply .github/workflows/cre-system-tests.yaml]: Thanks, updated.

File: system-tests/tests/regression/cre/cre_regression_suite_test.go (hunk 10 above)
Comment (chudilka1) [reply system-tests/tests/regression/cre/cre_regression_suite_test.go]: Updated.

File: system-tests/tests/regression/cre/README.md (hunk 11 above)
Comment (chudilka1) [reply system-tests/tests/regression/cre/README.md]: Thanks, updated.

File: system-tests/tests/go.mod (hunk 12 above)
Comment (chudilka1) [reply system-tests/tests/go.mod]: Done

File: system-tests/tests/go.mod (hunk 13 above)
Comment (chudilka1) [reply system-tests/tests/go.mod]: Done

File: .github/workflows/cre-system-tests.yaml (hunk 7 above)
Comment (chudilka1) [reply .github/workflows/cre-system-tests.yaml]: Moved

**Review by mchain0 — APPROVED** [review]

**Review by george-dorin — APPROVED** [review]

**Review by Tofel — APPROVED** [review]
